pages/emailSender.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState } from 'react'
+import {
+  ThemeProvider,
+  Divider,
+  Typography,
+  FormControl,
+  TableContainer,
+  Paper,
+  TableBody,
+  TableCell,
+  TableHead,
+  Button
+} from '@mui/material'
+import type { NextPage } from 'next'
+import { nanoid } from 'nanoid'
+import { useTheme } from '@mui/material/styles'
+import { StyledButton, StyledPageContainer } from '../styles/common'
+import { CsvRow, ReplaceObj, Message } from '../lib/types'
+import {
+  SectionContainer,
+  StyledCsvButton,
+  StyledTextArea,
+  StyledCsvButtonsContainer,
+  StyledSubHeader,
+  StyledFinalMessagesContainer,
+  StyledTableContainer,
+  StyledDivider,
+  StyledTable,
+  StyledTableRow
+} from '../pageStyles/emailSender.styles'
+
+const EmailSender: NextPage = () => {
+  const [file, setFile] = useState()
+  const [csvRowsArray, setCsvRowsArray] = useState<CsvRow[]>([])
+  const [message, setMessage] = useState('')
+  const [finalMessages, setFinalMessages] = useState<Message[]>([])
+  const theme = useTheme()
+
+  let reader: FileReader
+  if (typeof window !== 'undefined') {
+    reader = new window.FileReader()
+  }
+  const handleOnChange = (e: any) => {
+    setFile(e.target.files[0])
+  }
+
+  const csvFileToArray = (str: string) => {
+    const csvHeader = str.slice(0, str.indexOf('\n')).split(',')
+    const csvRows = str.slice(str.indexOf('\n') + 1).split('\n')
+    const array = csvRows.map((i) => {
+      const values = i.split(',')
+      const obj = csvHeader.reduce((object: any, header: any, index: any) => {
+        object[header] = values[index]
+        return object
+      }, {})
+      return obj
+    })
+
+    if (array[array.length - 1].email === '') {
+      array.pop()
+    }
+
+    setCsvRowsArray(array)
+  }
+
+  const handleOnSubmit = (e: any) => {
+    e.preventDefault()
+    if (file) {
+      reader.onload = function (event) {
+        const csvOutput = event.target?.result
+        if (typeof csvOutput === 'string') {
+          csvFileToArray(csvOutput)
+        }
+      }
+      reader.readAsText(file)
```

Comment:
are we displaying an error message anywhere if the csv file couldn't be parsed correctly? Or, are we relying on the user to look at each email to make sure everything worked? I'm not sure about what specific kinds of errors can pop up here hahah so leaving this up to you - something to think about

Reply:
thanks for bringing this up - will add csv validation as work for a future sprint based off ideas from today's brainstorming!